src/Adapter/MSTest.CoreAdapter/Discovery/TypeEnumerator.cs

```diff
@@ -192,11 +192,8 @@ internal UnitTestElement GetTestFromMethod(MethodInfo method, bool isDeclaredInT
                 testElement.Description = descriptionAttribute.Description;
             }
 
-            var workItemAttributeArray = this.reflectHelper.GetCustomAttributes(method, typeof(WorkItemAttribute)) as WorkItemAttribute[];
-            if (workItemAttributeArray != null)
-            {
-                testElement.WorkItemIds = workItemAttributeArray.Select(x => x.Id.ToString()).ToArray();
-            }
+            var workItemAttributeArray = this.reflectHelper.GetCustomAttributes(method, typeof(WorkItemAttribute)).Cast<WorkItemAttribute>();
+            testElement.WorkItemIds = workItemAttributeArray.Select(x => x.Id.ToString()).ToArray();
```

Comment:
We're not doing a null check here, What happens if `workItemAttributeArray` is empty?

Reply:
- The `GetCustomAttributes` method is simply mapped to `MemberInfo.GetCustomAttributes()`, which doesn't return null ([docs](https://docs.microsoft.com/en-us/dotnet/api/system.reflection.memberinfo.getcustomattributes?view=netframework-4.5.1)). 
- As we are not expecting the `GetCustomAttributes` method to return any `Attribute` other than `WorkItemAttribute`, the `.Cast()` extension method is the right choice here. It returns the casted enumerable or throws if any object is not of the specified type.

So a null check as it was implemented before doesn't make sense anymore.

If the `WorkItemAttribute` is empty, the `testElement.WorkItemIds` property is set to an empty array. IMO this makes sense, but then I remembered the following snippet in the `UnitTestElement` class:

``` cs
if (this.WorkItemIds != null)
{
    testCase.SetPropertyValue(TestAdapter.Constants.WorkItemIdsProperty, this.WorkItemIds);
}
```

So, we found a bug here: The `UnitTestElement` class sets the test case property only if the `WorkItemIds` property is null, which cannot happen anymore in this implementation.

In order to fix the bug and stick with the existing conventions, I altered the implementation to set the `WorkItemIds` property to null if no WorkitemIds exist. I added the UnitTest `GetTestFromMethodShouldSetWorkItemIdsToNullIfNotAny` to ensure this behavior is not altered.

Reply:
Thanks for the fix. I'll try to approve this PR today _or tomorrow._